Data/IntMap/Internal.hs

```diff
@@ -2091,27 +2094,39 @@ deleteFindMax = fromMaybe (error "deleteFindMax: empty map has no maximal elemen
 deleteFindMin :: IntMap a -> ((Key, a), IntMap a)
 deleteFindMin = fromMaybe (error "deleteFindMin: empty map has no minimal element") . minViewWithKey
 
--- | /O(min(n,W))/. The minimal key of the map.
-findMin :: IntMap a -> (Key, a)
-findMin Nil = error $ "findMin: empty map has no minimal element"
-findMin (Tip k v) = (k,v)
-findMin (Bin _ m l r)
+-- | /O(min(n,W))/. The minimal key of the map. Returns 'Nothing' if the map is empty.
+lookupMin :: IntMap a -> Maybe (Key, a)
+lookupMin Nil = Nothing
+lookupMin (Tip k v) = Just (k,v)
+lookupMin (Bin _ m l r)
   | m < 0     = go r
   | otherwise = go l
-    where go (Tip k v)      = (k,v)
+    where go (Tip k v)      = Just (k,v)
           go (Bin _ _ l' _) = go l'
-          go Nil            = error "findMax Nil"
+          go Nil            = Nothing
```

Comment:
We have to decide whether the non-`Nil` case should produce `Just` immediately, or wait till the descent is complete. I believe that @wrengr prefers the former and that we do so for `Map`. I have some reservations about that approach from the potential space leak angle, but whichever way we choose we should be consistent.

Reply:
Would it make sense to have both versions, one in Lazy and one in Strict?

Reply:
I don't think that's a very good fit, no. I'd like if @wrengr would remind me why the lazy way might ever be a good idea. I can't remember or think of a really good reason for it right now.